numba/cpython/unicode.py

```diff
@@ -842,6 +830,56 @@ def _adjust_indices(length, start, end):
     return start, end
 
 
+@overload_method(types.UnicodeType, 'startswith')
+def unicode_startswith(s, prefix, start=None, end=None):
+    if not (start is None or isinstance(start, (types.Omitted,
+                                                types.Integer,
+                                                types.NoneType))):
+        raise TypingError('When specified, the arg "start" must be an Integer')
```

Comment:
I'm not sure this error message is correct? `None` is accepted so it would be entirely legal for a user to write:
```python
'abcdef'.startswith('f', None)
```
which is not an integer.

Reply:
Revised as expected.